configurator/src/main/java/com/devonfw/tools/ide/migrator/Migrations.java

```diff
@@ -197,4 +197,28 @@ public static MigrationImpl devon4j() {
         .next().build();
   }
 
+  public static MigrationImpl junit() {
+    return new MigrationBuilder().to(null)
+        .pom()
+        .replaceString("com.devonfw.module.test.common.api.category.CategorySystemTest", "system")
+        .replaceString(
+            "com.devonfw.module.test.common.api.category.CategoryComponentTest,com.devonfw.module.test.common.api.category.CategorySubsystemTest,com.devonfw.module.test.common.api.category.CategorySystemTest",
+            "component,subsystem,system")
+        .replaceString(
+            "com.devonfw.module.test.common.api.category.CategorySubsystemTest,com.devonfw.module.test.common.api.category.CategorySystemTest",
+            "subsystem,system")
+        .replaceDependency(new VersionIdentifier("junit", "junit", null),
+            new VersionIdentifier("org.junit.jupiter", "junit-jupiter-engine", null, VersionIdentifier.SCOPE_TEST))
+        .addDependency(new VersionIdentifier("*-core", null),
+            new VersionIdentifier("org.junit.platform", "junit-platform-runner", null, VersionIdentifier.SCOPE_TEST))
+        .and().java()
+        .replace("import org.junit.Test;",
+            "import org.junit.jupiter.api.extension.ExtendWith;\n import org.junit.jupiter.api.Test;\n import org.springframework.test.context.junit.jupiter.SpringExtension;")
+        .replace("@RunWith(SpringRunner.class)", "@ExtendWith(SpringExtension.class)")
+        .replace("com.devonfw.module.jpa.dataaccess.api.RevisionMetadata",
+            "com.devonfw.module.basic.common.api.RevisionMetadata")
+        .replace("com.devonfw.module.jpa.dataaccess.api.RevisionMetadataType",
+            "com.devonfw.module.basic.common.api.RevisionMetadataType")
+        .and().next().build();
```

Comment:
nice starting point. However a migration from junit 4 to 5 should not be strictly limited to devon4j spring projects.
However, we can merge this and later add PR for improvements. Actually this migration is rather tricky. We could find several projects at CG to test and improve this...